src/codecs/dxt.rs

```diff
@@ -525,16 +525,21 @@ fn encode_dxt_colors(source: &[u8], dest: &mut [u8], is_dxt1: bool) {
         *d = [s[0], s[1], s[2]];
     }
 
-    // and a set of colors to pick from.
-    let mut colorspace = targets.to_vec();
-
     // roundtrip all colors through the r5g6b5 encoding
-    for rgb in &mut colorspace {
+    for rgb in &mut targets {
         *rgb = enc565_decode(enc565_encode(*rgb));
     }
 
     // and deduplicate the set of colors to choose from as the algorithm is O(N^2) in this
-    colorspace.dedup();
+    let mut colorspace_ = [[0u8; 3]; 16];
+    let mut colorspace_len = 0;
+    for color in &targets {
+        if !colorspace_.contains(color) {
```

Comment:
`dedup` is not the same as keeping only unique colors and this will fail to include `[0, 0, 0]` if it ever occurs. I don't know enough about `tga` to know for certain if this behavioral difference affects the actual encoding correctness.